### PR TITLE
Presets / Addon-docs: Cleanup framework-specific presets

### PR DESCRIPTION
**Pull request by shilman**
Issue: N/A

Pre-work for #5731

## What I did

- [x] Add a `framework` option to presets
- [x] Update `addon-docs` to use `framework` option, unifying the docs preset
- [x] Deprecate `addon-docs` framework-specific presets
- [x] Update documentation

## How to test

Try out:
- official-storybook
- vue-kitchen-sink
- angular-cli
- etc.

### CODE REVIEW

File: addons/docs/src/frameworks/common/makePreset.ts [hunk 1]
```diff
@@ -1,25 +1,17 @@
-import fs from 'fs';
-import * as common from './preset';
+import deprecate from 'util-deprecate';
+import dedent from 'ts-dedent';
+import * as common from '../../preset';
```
Comment (mrmckeb) [inline addons/docs/src/frameworks/common/makePreset.ts]: You can avoid this by enabling ESM interop in TypeScript.

File: addons/docs/src/frameworks/common/preset.ts [hunk 2]
```diff
@@ -72,8 +77,20 @@ export function webpack(webpackConfig: any = {}, options: any = {}) {
       ],
     },
   };
+  return result;
 }
 
-export function addons(entry: any[] = []) {
+export function addons(entry: any[] = [], options: any) {
```
Comment (mrmckeb) [inline addons/docs/src/frameworks/common/preset.ts]: We need to look at some global types for options, etc.

File: addons/docs/src/frameworks/vue/preset.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export function webpack(webpackConfig: any = {}, options: any = {}) {
```
Comment (mrmckeb) [inline addons/docs/src/frameworks/vue/preset.ts]: There is a type for Webpack config, `Configuration` from `webpack`.

**Review by mrmckeb — APPROVED** [review]
Looks great, it would be great to get rid of some of the any's here, but I think that's a bigger task.

**Review by ndelangen — APPROVED** [review]

File: addons/docs/src/frameworks/vue/preset.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  webpackConfig.module.rules.push({
```
Comment (igor-dv) [inline addons/docs/src/frameworks/vue/preset.ts]: better to make this pure.

**Review by igor-dv — APPROVED** [review]

**Pull request merged**